src/util/manipulator_util.ts

```diff
@@ -0,0 +1,163 @@
+/*
+    Copyright 2020 Google LLC
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+import {Node, ts, Type, Statement, TypeFormatFlags} from 'ts-morph';
+
+export class ManipulatorUtil {
+  /**
+   * Verifies that a node hasn't been edited before by looking through leading comments and
+   * ensuring that comments weren't left by previous iterations of fixes.
+   * @param {Node<ts.Node>} node - Node to verify.
+   * @param {string} comment - Comment to look for when parsing leading comment ranges of node.
+   * @return {boolean} True if node hasn't been editted before.
+   */
+  static verifyCommentRange(node: Node<ts.Node>, comment: string): boolean {
+    return !node.getLeadingCommentRanges().some(commentRange => {
+      return commentRange.getText().includes(comment);
+    });
+  }
+
+  /**
+   * Converts a Union type into a list of base types, if applicable.
+   * @param {Type} type - Input type.
+   * @return {Type[]} List of types represented by input type.
+   */
+  static toTypeList(type: Type): Type[] {
+    return type.isUnion()
+      ? type.getUnionTypes().map(individualType => {
+          return individualType.getBaseTypeOfLiteralType();
+        })
+      : [type.getBaseTypeOfLiteralType()];
+  }
+
+  /**
+   * Returns a modified node's closest Statement ancestor, or the node itself if it is Statement.
+   * @param {Node<ts.Node>} node - Modified node.
+   * @return {Statement|undefined} Closest Statement ancestor of modified node or undefined if doesn't exist.
+   */
+  static getModifiedStatement(node: Node<ts.Node>): Statement | undefined {
+    if (Node.isStatement(node)) {
+      return node;
+    }
+
+    return node.getParentWhile((parent, child) => {
+      return !(Node.isStatementedNode(parent) && Node.isStatement(child));
+    }) as Statement | undefined;
+  }
+
+  /**
+   * Returns if type is valid. Currently, any type with "any" and "never" is invalid.
+   * @param {string} type - Type to be evaluated.
+   * @return {boolean} True if type is valid.
+   */
+  static isValidType(type: string): boolean {
+    return !type.includes('any') && !type.includes('never');
+  }
+
+  /**
+   * Returns the string representation of a type.
+   * @param {Type<ts.Type>} type - Type to be converted to a string.
+   * @return {string[]} String representation of type.
+   */
+  static typeToString(
```

Comment:
may be add a simple test to other function as well?

Reply:
Good point, I'll add unit tests for both these pure functions and StrictNullChecksManipulator